website/docs/getting-started/project-walkthrough.md

```diff
@@ -0,0 +1,169 @@
+---
+sidebar_position: 2
+---
+
+# Project Walkthrough
+
+The template generates several files, most notably:
+
+- `./package.json` includes `scripts` to `validate`, `synth`, and `deploy` your project.
+- `./tsconfig.json` configures the `@functionless/language-service` plugin to improve IDE feedback.
+- `./cdk.json` points CDK to the application entrypoint and requires the `functionless/register` hook to transpile project files.
+- `./src/app.ts` is the appliction entrypoint.
+
+Let's take a closer look at `./src/app.ts`:
+
+```typescript
+import { App, Stack } from "aws-cdk-lib";
+import { Function, StepFunction } from "functionless";
+
+const app = new App();
+
+const stack = new Stack(app, "MyStack");
+
+const sayFunction = new Function(
+  stack,
+  "SayFunction",
+  async (event: { message: string }) => {
+    console.log(event.message);
+    return;
+  }
+);
+
+new StepFunction(stack, "Workflow", async (event: { name: string }) => {
+  await sayFunction({ message: `Hello ${event.name}` });
+});
+```
+
+We start by importing Constructs from `aws-cdk-lib` and `functionless`:
+
+```typescript
+import { App, Stack } from "aws-cdk-lib";
+import { Function, StepFunction } from "functionless";
+```
+
+:::note
+
+Most Constructs wrap some underlying Resource and provide a high-level interface for configuring the service that backs the Resource.
+
+You can read more about [Constructs][cdk-construct] and [Resources][cdk-resource] in the AWS CDK documentation.
+
+:::
+
+Functionless exposes AWS CDK compatible Resource Constructs from its top-level API.
+In this example, we're using the `Function` and `StepFunction` Resource Constructs which map to the AWS Lambda and AWS Step Functions services.
+
+Next, we initialize `app` which will be the root of our CDK Construct tree:
+
+```typescript
+const app = new App();
+```
+
+CDK models infrastructure as a tree of Constructs with a single instance of `App` at the root.
+To add Constructs to the tree, we pass the parent as the first argument when instantiating a child.
```

Comment:
The CDK terminology is `scope`. Do we think `parent` will resonate more?

Reply:
I started with scope but I found it hard to write about in the context of helping beginners build a mental model. Will have to review the CDK docs again. Probably a good idea to be consistent with them.

Reply:
I can also see an argument for why parent is easier to understand when talking about trees